net/FlatBuffers/FlatBufferBuilder.cs

```diff
@@ -317,10 +353,47 @@ public void Add<T>(T[] x)
             int size = ByteBuffer.SizeOf<T>();
             // Need to prep on size (for data alignment) and then we pass the
             // rest of the length (minus 1) as additional bytes
-            Prep(size, size * (x.Length - 1));
+            Prep(size, size * (x.Count - 1));
             Put(x);
         }
 
+        /// <summary>
+        /// Adds the data of type T pointed to by the given pointer to the buffer (aligns the data and grows if necessary).
+        /// </summary>
+        /// <typeparam name="T">The type of the input data</typeparam>
+        /// <param name="ptr">The pointer to copy data from</param>
+        /// <param name="sizeInBytes">The data size in bytes</param>
+        public void Add<T>(IntPtr ptr, int sizeInBytes)
+            where T : struct
+        {
+            if (ptr == IntPtr.Zero)
+            {
+                throw new ArgumentNullException("Cannot add a null pointer");
+            }
+
+            if(sizeInBytes == 0)
```

Comment:
Maybe move this up higher, so you don't throw in the case of 0 bytes?

Reply:
Sure, will do that. I'll add another missing sanity check underneath:
`if(sizeInBytes < 0)
{
    throw new ArgumentOutOfRangeException("sizeInBytes", "sizeInBytes cannot be negative");
}`